packages/astro/src/compiler/codegen/index.ts

```diff
@@ -249,7 +249,7 @@ function compileModule(module: Script, state: CodegenState, compileOptions: Comp
           } else if (node.declaration.type === 'FunctionDeclaration') {
             // case 2: createCollection (export async function)
             if (!node.declaration.id || node.declaration.id.name !== 'createCollection') break;
-            createCollection = module.content.substring(node.declaration.start || 0, node.declaration.end || 0);
+            createCollection = module.content.substring(node.start || 0, node.end || 0);
```

Comment:
This was the source of the error: we were accidentally stripping out the `export` statement when hoisting the function. Somehow we noticed this, and added it back, but only in specific cases (hence the `fetchContent()` being necessary). This was fixed and is no longer the case.

Reply:
nice! great catch